src/key_events.cpp

```diff
@@ -40,57 +43,137 @@ static bool handleKeyswitchEventDefault(Key mappedKey, byte row, byte col, uint8
+    if (mappedKey.raw == Key_NoKey.raw)
+      return;
+  }
+  mappedKey = Layer.eventHandler(mappedKey, row, col, keyState);
+  if (mappedKey.raw == Key_NoKey.raw)
+    return;
+  handleKeyswitchEventDefault(mappedKey, row, col, keyState);
+}
+
+void pressKeyRaw(Key mappedKey) {
+  Keyboard.press(mappedKey.keyCode);
+
+}
+
 void pressKey(Key mappedKey) {
   if (mappedKey.flags & SHIFT_HELD) {
-    Keyboard.press(Key_LeftShift.keyCode);
+    pressKeyRaw(Key_LeftShift);
   }
   if (mappedKey.flags & CTRL_HELD) {
-    Keyboard.press(Key_LeftControl.keyCode);
+    pressKeyRaw(Key_LeftControl);
   }
   if (mappedKey.flags & LALT_HELD) {
-    Keyboard.press(Key_LeftAlt.keyCode);
+    pressKeyRaw(Key_LeftAlt);
   }
   if (mappedKey.flags & RALT_HELD) {
-    Keyboard.press(Key_RightAlt.keyCode);
+    pressKeyRaw(Key_RightAlt);
   }
   if (mappedKey.flags & GUI_HELD) {
-    Keyboard.press(Key_LeftGui.keyCode);
+    pressKeyRaw(Key_LeftGui);
   }
-  Keyboard.press(mappedKey.keyCode);
+
+  pressKeyRaw(mappedKey);
+}
+
+void releaseKeyRaw(Key mappedKey) {
+  Keyboard.release(mappedKey.keyCode);
+
 }
 
+void releaseAllKeys() {
+  Keyboard.releaseAll();
+}
 
 void releaseKey(Key mappedKey) {
   if (mappedKey.flags & SHIFT_HELD) {
-    Keyboard.release(Key_LeftShift.keyCode);
+    releaseKeyRaw(Key_LeftShift);
   }
   if (mappedKey.flags & CTRL_HELD) {
-    Keyboard.release(Key_LeftControl.keyCode);
+    releaseKeyRaw(Key_LeftControl);
   }
   if (mappedKey.flags & LALT_HELD) {
-    Keyboard.release(Key_LeftAlt.keyCode);
+    releaseKeyRaw(Key_LeftAlt);
   }
   if (mappedKey.flags & RALT_HELD) {
-    Keyboard.release(Key_RightAlt.keyCode);
+    releaseKeyRaw(Key_RightAlt);
   }
   if (mappedKey.flags & GUI_HELD) {
-    Keyboard.release(Key_LeftGui.keyCode);
+    releaseKeyRaw(Key_LeftGui);
   }
-  Keyboard.release(mappedKey.keyCode);
+  releaseKeyRaw(mappedKey);
 }
 
-void handleKeyswitchEvent(Key mappedKey, byte row, byte col, uint8_t keyState) {
-  if (!(keyState & INJECTED)) {
-    mappedKey = Layer.lookup(row, col);
-  }
-  for (byte i = 0; Kaleidoscope.eventHandlers[i] != NULL && i < HOOK_MAX; i++) {
-    Kaleidoscope_::eventHandlerHook handler = Kaleidoscope.eventHandlers[i];
-    mappedKey = (*handler)(mappedKey, row, col, keyState);
-    if (mappedKey.raw == Key_NoKey.raw)
-      return;
-  }
-  mappedKey = Layer.eventHandler(mappedKey, row, col, keyState);
-  if (mappedKey.raw == Key_NoKey.raw)
-    return;
-  handleKeyswitchEventDefault(mappedKey, row, col, keyState);
+
+
+
+void sendKeyboardReport() {
+  Keyboard.sendReport();
+}
+
+void pressConsumer(Key mappedKey) {
+  ConsumerControl.press(mappedKey.keyCode);
+}
+
+void releaseConsumer(Key mappedKey) {
+  ConsumerControl.release(mappedKey.keyCode);
+}
+
+void pressSystem(Key mappedKey) {
+  SystemControl.press(mappedKey.keyCode);
+}
+
+void releaseSystem(Key mappedKey) {
+  SystemControl.release();
+}
+
+
+/** Mouse events
+ * See above for commentary on connectionMask. */
```

Comment:
`connectionMask`? Are those the buttons? (I do not see a comment about this thing added)